### PR TITLE
feat: added text area component with corresponding docs

### PR DESCRIPTION
**Pull request by justinfaynhan**

### CODE REVIEW

**Review by matildaPan — APPROVED** [review]
🎉

File: packages/text-area/src/TextArea.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  const theme = useTheme();
+  const inputStyles = useInput({
+    disabled,
+    invalid,
+  });
+
+  return {
+    ...inputStyles,
+    overflow: 'hidden', // fix for Safari to prevent unwanted scrolling of parent container to occur
+    textOverflow: 'ellipsis',
```
Comment (lukebennett88) [inline packages/text-area/src/TextArea.tsx:77]: We can remove these, they were specific for the `Select`

File: packages/text-area/src/TextArea.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    resize: 'vertical',
+    minHeight: theme.sizing.medium,
+
+    // Prevent text going underneath the chevron icon
+    paddingRight:
+      theme.sizing.xxsmall + // size of chevron icon
+      theme.spacing.medium * 2, // paddingX value
```
Comment (lukebennett88) [inline packages/text-area/src/TextArea.tsx:84]: Same as above. This was reserving space for the chevron icon for the `Select` so we don't need it.

**Review by lukebennett88 — CHANGES_REQUESTED** [review]

File: packages/text-area/README.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        <TextArea value={textInput} onChange={event => setTextInput(event.target.value)}/>
+      </Field>
+      <Text>You have inputed: </Text>
+      {textInput && (
+        <Text>{textInput}</Text>
+      )}
+    </Stack>
+  )
+```
+
```
Comment (lukebennett88) [inline packages/text-area/README.md:32]: Happy to approve this as is, but feels a little lite. Would be nice to flesh this out a tiny bit more.

[Balance has a size prop](https://balance.reckon.com/package/textarea#size) @jordangeizer do you think we should do the same?

Comment (justinfaynhan) [reply packages/text-area/README.md:34]: Added more doc examples regarding text area demonstrating size and weight use cases

**Review by lukebennett88 — APPROVED** [review]

**Pull request merged**